lib/src/analysis_options.dart

```diff
@@ -108,5 +108,15 @@ String updatePassthroughOptions({
     }
   }
 
+  final origFormatter = origMap['formatter'];
+  if (origFormatter is Map) {
+    final pageWidth = origFormatter['page_width'];
+    if (pageWidth is int) {
+      final customFormatter =
+          customMap.putIfAbsent('formatter', () => <String, Object?>{}) as Map;
+      customFormatter['page_width'] = pageWidth;
+    }
+  }
```

Comment:
FYI this could also be:

```dart
  if (origMap['formatter'] case {'page_width': int pageWidth}) {
    final customFormatter =
        customMap.putIfAbsent('formatter', () => <String, Object?>{}) as Map;
    customFormatter['page_width'] = pageWidth;
  }
```

Reply:
@lrhn only if you use dart format with a page width of at-least 103 chars :rofl: 

Otherwise it becomes (once formatted):
```dart
  if (origMap['formatter'] case {'page_width': int pageWidth}) {
    (customMap.putIfAbsent('formatter', Map<String, Object?>.empty)
        as Map)['page_width'] = pageWidth;
  }
```

Personally, I'd suggest it's more readable (albeit a bit dumb) to do:
```dart
  if (origMap['formatter'] case {'page_width': int pageWidth}) {
    customMap['formatter'] ??= <String, Object?>{};
    customMap['formatter']['page_width'] = pageWidth;
  }
```

or 
```dart
  if (origMap['formatter'] case {'page_width': int pageWidth}) {
    customMap['formatter'] = {
      ...?customMap['formatter'],
      'page_width': pageWidth,
    };
  }
```
Arguably, they aren't as efficient, but they read nicely -- which is sadly almost always the most important thing (at-least in `pana`).